gobblin-data-management/src/main/java/org/apache/gobblin/runtime/retention/DatasetCleanerTask.java

```diff
@@ -47,8 +51,15 @@ public void run() {
       DatasetCleaner datasetCleaner = new DatasetCleaner(FileSystem.get(new Configuration()),
           this.taskContext.getTaskState().getProperties());
       datasetCleaner.clean();
+      this.workingState = WorkUnitState.WorkingState.SUCCESSFUL;
     } catch (IOException e) {
+      this.workingState = WorkUnitState.WorkingState.FAILED;
       throw new RuntimeException(e);
     }
   }
+
+  @Override
+  public void commit() {
+    log.info("task {} commits with state {}", this.taskContext.getTaskState().getTaskId(), this.workingState);
```

Comment:
Just curious, if this task runs in mapper or driver? I think most gobblin retention job runs on driver, if that's the case, what's the taskID here?

Reply:
regardless where it runs, a task will have a task id, right?